esphome/components/image/__init__.py

```diff
@@ -74,6 +77,56 @@ def download_mdi(value):
     return value
 
 
+def get_filename_from_url(url):
+    # Regular expression pattern to match the file name at the end of the URL
+    pattern = r"/([^/]+)$"
```

Comment:
You can put two subgroups in the regex, and then parse filename and extension with a single call.
```suggestion
    pattern = r"/([^/]+)(\.([^./]+))?$"
```
`group(1)` would be filename (without extension) and `group(3)`  the extension (only if group(2) finds a match).
Then you could return a tuple instead of a single value, and avoid the next method.